stonesoup/metricgenerator/tracktotruthmetrics.py

```diff
@@ -207,7 +214,11 @@ def S_time_range(self, manager):
             self._n_t(manager, timestamp) - self._na_t(manager, timestamp)
             for timestamp in timestamps)
 
-        S = numerator / self._n_sum(manager, timestamps)
+        try:
+            S = numerator / self._n_sum(manager, timestamps)
+        except ZeroDivisionError:
+            # No tracks
+            S = 0
```

Comment:
Its not as clear cut as just returning Zero when the division fails. S is a measure of percentage of spurious tracks, returning 0 indicates perfect performance where every true track is associated with an object which is being tracked. In this case it should probably return an error message to say that there were no tracks

Reply:
So I think raising an error needs to be avoided here, as wouldn't want to have to handle that when processing a batch run. Given you'd be looking at set of metrics as a whole, I don't see an issue with reporting that number of spurious tracks is zero, when number of tracks is also zero (which seems like valid value for the metric in that scenario).

Reply:
I've updated the code to produce warnings, and hopefully the outputs are sensible in context of the warnings.